drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/controller/Events.java

```diff
@@ -43,21 +43,23 @@ public Events( final EventBus eventBus,
     }
 
     public void setup() {
-
-        eventBusHandlerRegistrations.add( eventBus.addHandler( ValidateEvent.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( DeleteRowEvent.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( AfterColumnDeleted.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( UpdateColumnDataEvent.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( AppendRowEvent.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( InsertRowEvent.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( AfterColumnInserted.TYPE,
-                                                               analyzerController ) );
+        if ( canSetUp ) {
```

Comment:
Right, I can set if the set is empty and remove this boolean. Good point, the day has been too long. Will fix this.